### PR TITLE
Dem output solid object option

### PR DESCRIPTION
**Pull request by OGaboriault**
# Description of the problem

- Solid object output files were always being outputted at every output time step. When using a large amount of solid object, the amount of output files can become astronomical pretty quickly. 

# Description of the solution

- I have added a new parameter to solid object. We can now specify if we want output files to be generated or not. By default, this parameter is at true, which keep compatibility with the previous parameter files (like the examples). 

# How Has This Been Tested?

- I used one of my prm file for my LPBF simulation and put solid object 0 and 2 to false. Ran the simulation with zero particle and no output file were generated for those two solid objects. 

# Documentation

- Small correction to the solid object documentation.
- Added the parameter to the doc.

# Future changes

- We could replace this new bool to a function like for the "translational velocity", so that vtus would be outputted during a certain period.

### CODE REVIEW

File: doc/source/parameters/dem/floating_mesh.rst [hunk 1]
```diff
@@ -45,6 +45,10 @@ This subsection explains the solid objects (floating meshes) information. First
             # Z COR
             set z = 0
         end
+
+        subsection output
+            set output solid object = true
+        end
```
Comment (blaisb) [inline doc/source/parameters/dem/floating_mesh.rst:51]: Why did you to create a subsection? Could it not be just a free parameter?

File: include/core/serial_solid.h [hunk 2]
```diff
@@ -291,7 +291,8 @@ class SerialSolid
   Vector<double>                                displacement_since_mapped;
 
   // Output management
-  PVDHandler pvdhandler;
+  PVDHandler  pvdhandler;
+  const bool *output_bool;
```
Comment (blaisb) [inline include/core/serial_solid.h:295]: why a pointer???

File: source/core/serial_solid.cc [hunk 3]
```diff
@@ -48,6 +45,7 @@ SerialSolid<dim, spacedim>::SerialSolid(
   , this_mpi_process(Utilities::MPI::this_mpi_process(mpi_communicator))
   , param(param)
   , id(id)
+  , output_bool(&param->output_bool)
```
Comment (blaisb) [inline source/core/serial_solid.cc:48]: no need for a pointer with an adress, just pass the value itself. I don't see the point of having this pool pointer floating around

File: source/core/serial_solid.cc [hunk 4]
```diff
@@ -469,52 +467,55 @@ void
...
-
-  write_vtu_and_pvd<dim, spacedim>(pvdhandler,
-                                   data_out,
-                                   folder,
-                                   solution_name,
-                                   time,
-                                   iter,
-                                   group_files,
-                                   this->mpi_communicator);
+  if (*output_bool)
```
Comment (blaisb) [inline source/core/serial_solid.cc:470]: instead I would have something like this
```suggestion
  if (!*output_bool) return;
```
You just leave the function if you are not outputting. Easier to read

**Review by blaisb — CHANGES_REQUESTED** [review]
Some comments to address, The pointer thing is very weird

File: doc/source/parameters/dem/floating_mesh.rst (hunk 1 above)
Comment (OGaboriault) [reply doc/source/parameters/dem/floating_mesh.rst:51]: Yes it can be. It would be the only parameter not in a subsection. It's not a problem in itself. ]

**Review by acdaigneault — COMMENTED** [review]
My comments are the same as Bruno (pointer & subsections)

**Review by blaisb — APPROVED** [review]
Much better, thanks!

**Pull request merged**